src/core/router/net_db/impl.cc

```diff
@@ -163,30 +165,14 @@ void NetDb::Run() {
       }
       // builds exploratory tunnels at Nth interval to find more peers
       if (ts - last_exploratory >= static_cast<std::uint16_t>(NetDbInterval::Exploratory)) {
-        // Set default exploratory count
-        std::uint16_t exploratory_count =
-          static_cast<std::uint16_t>(NetDbSize::MinExploratoryTunnels);
-        // Get number of current available routers
         auto known_routers = GetNumRouters();
-        // Evaluates if a router has a sufficient number of known routers
-        // to use for building tunnels and sets exploratory count as needed
-        if (known_routers <
-            static_cast<std::uint16_t>(NetDbSize::FavouredKnownRouters)
-            || ts - last_exploratory >=
-            static_cast<std::uint16_t>(NetDbInterval::DelayedExploratory)) {
-          // Test if we're below the desired threshold
-          if (known_routers <
-              static_cast<std::uint16_t>(NetDbSize::MinKnownRouters)) {
-            // Set the max exploratory count
-            exploratory_count =
-              static_cast<std::uint16_t>(NetDbSize::MaxExploratoryTunnels);
-          }
-        }
+        std::uint16_t exploratory_count = UpdateNumExploratoryTunnels(
```

Comment:
Consider `auto` for type safety / implicit conversion, e.g., if we change the return type for `UpdateNumExploratoryTunnels` to `std::uint32_t` or larger, don't count on remembering to refactor `exploratory_count` unless you have a solid design in place.